client/my-sites/checkout/gsuite-nudge/index.jsx

```diff
@@ -36,29 +39,50 @@ export class GSuiteNudge extends React.Component {
 		selectedSiteId: PropTypes.number.isRequired,
 	};
 
+	isMounted = false;
+
+	componentDidMount() {
+		this.isMounted = true;
+	}
+
+	componentWillUnmount() {
+		this.isMounted = false;
+	}
+
 	handleSkipClick = () => {
-		this.props.handleCheckoutCompleteRedirect();
+		const getThankYouPageUrlArguments = {
+			siteSlug: this.props.siteSlug,
+			receiptId: this.props.receiptId,
+			cart: this.props.cart,
+		};
+		const url = getThankYouPageUrl( getThankYouPageUrlArguments );
+		page.redirect( url );
 	};
 
 	handleAddEmailClick = ( cartItems ) => {
-		const { siteSlug, receiptId } = this.props;
+		const { siteSlug, receiptId, productsList } = this.props;
 		this.removePlanFromCart();
 
-		addItems(
-			// add `receipt_for_domain` to cartItem extras
-			cartItems.map( ( item ) => ( {
-				...item,
-				extra: { ...item.extra, receipt_for_domain: receiptId },
-			} ) )
-		);
-
-		page( `/checkout/${ siteSlug }` );
+		this.props.shoppingCartManager
+			.addProductsToCart(
+				// add `receipt_for_domain` to cartItem extras
+				cartItems
+					.map( ( item ) => ( {
+						...item,
+						extra: { ...item.extra, receipt_for_domain: receiptId },
+					} ) )
+					.map( ( item ) => fillInSingleCartItemAttributes( item, productsList ) )
+			)
+			.then( () => {
+				this.isMounted && page( `/checkout/${ siteSlug }` );
+			} );
 	};
 
-	removePlanFromCart() {
+	removePlanFromCart = () => {
```

Comment:
Upon further inspection maybe the plain is supposed to be removed. :)

Reply:
Yes, I think it is. But I think I actually translated this logic backwards.

Reply:
But if you still saw the plan removed, that implies that it was happening on the server, so this code may in fact be unnecessary. I'll verify.

Reply:
You know, I think it is unused because you're not likely to get to this upsell unless the cart is empty (as you've just made a purchase).